tests/unit/core/components/TestInstallButton.js

```diff
@@ -34,11 +34,13 @@ describe('<InstallButton />', () => {
     getClientCompatibility: () => ({ compatible: true }),
     hasAddonManager: true,
     i18n: getFakeI18nInst(),
+    store: createStore().store,
     ...customProps,
   });
 
-  const render = (props) =>
-    shallow(<InstallButtonBase {...renderProps(props)} />);
+  const render = (props) => shallowUntilTarget(
+    <InstallButton {...renderProps(props)} />, InstallButtonBase
```

Comment:
The only other option that's sort of like option 2 would be to setup a convention of declaring each HOC in an array we export as a property of each component which would clean up the test API, eg:

```js
// src/amo/components/Search.js
export class SearchBase extends React.Component {
  [...]
}

export const HigherOrderComponents = [
  connect(mapStateToProps),
  translate(),
];

const Search = compose(...HigherOrderComponents)(SearchBase);

Search._higherOrderComponents = HigherOrderCompoents;

export default Search;
```

But writing that out it's way more boilerplate and annoyance than just specifying `shallowUntilTarget(<Search>, SearchBase)`. So let's stick with your existing API!